test/system/src/test/java/io/pravega/test/system/MultiReaderWriterWithFailOverTest.java

```diff
@@ -0,0 +1,307 @@
+
+        readerGroupManager = ReaderGroupManager.withScope(scope, controllerUri);
+        readerGroupManager.createReaderGroup(readerGroupName, ReaderGroupConfig.builder().startingTime(0).build(),
+                Collections.singleton(STREAM_NAME));
+
+        log.info(" reader group name {} ", readerGroupManager.getReaderGroup(readerGroupName).getGroupName());
+        log.info(" reader group scope {}", readerGroupManager.getReaderGroup(readerGroupName).getScope());
+        //create 20 readers
+        log.info("creating {} readers", NUM_READERS);
+        List<CompletableFuture<Void>> readerList = new ArrayList<>();
+        String readerName = "reader" + new Random().nextInt(Integer.MAX_VALUE);
+        //start reading events
+        for (int i = 0; i < NUM_READERS; i++) {
+            log.info("starting reader{}", i);
+            readerList.add(startReader(readerName + i, clientFactory, readerGroupName,
+                    eventsReadFromPravega, eventData, eventReadCount, stopReadFlag));
+        }
+
+        Thread.sleep(50);
+
+        //wait for writers completion
+        for (int i = 0; i < writerList.size(); i++) {
+            log.info("get on writer list {} ", writerList.get(i).get());
+            CompletableFuture.allOf(writerList.get(i));
+        }
+
+        while (NUM_EVENTS != eventsReadFromPravega.size()) {
+            Thread.sleep(5);
+        }
+
+        stopReadFlag.set(true);
+
+        //wait for readers completion
+        for (int i = 0; i < readerList.size(); i++) {
+            log.info("get on readerlist {} ", readerList.get(i).get());
+            CompletableFuture.allOf(readerList.get(i));
+        }
+
+        log.info("All writers have stopped. Setting Stop_Read_Flag. Event Written Count:{}, Event Read " +
+                "Count: {}", eventData.get(), eventsReadFromPravega.size());
+        assertEquals(NUM_EVENTS, eventsReadFromPravega.size());
+        assertEquals(NUM_EVENTS, new TreeSet<>(eventsReadFromPravega).size()); //check unique events.
+        //stop reading when no. of reads= no. of writes
+        log.debug("test {} succeed", "multiReaderWriterTest");
+
+        //seal a stream
+        CompletableFuture<Boolean> sealsStreamStatus = controller.sealStream(scope, STREAM_NAME);
+        assertTrue(sealsStreamStatus.get());
+
+        //deleting stream
+        CompletableFuture<Boolean> deleteStreamStatus = controller.deleteStream(scope, STREAM_NAME);
+        assertTrue(deleteStreamStatus.get());
+
+    }
+
+    private CompletableFuture<Void> startNewWriter(final AtomicLong data,
+                                                   final ClientFactory clientFactory) {
+        return CompletableFuture.runAsync(() -> {
+            final EventStreamWriter<Long> writer = clientFactory.createEventWriter(STREAM_NAME,
+                    new JavaSerializer<Long>(),
+                    EventWriterConfig.builder().build());
+            for (int i = 0; i < 1000; i++) {
```

Comment:
Should we have another static variable for this number of events and make `NUM_EVENTS` a function of that value?